sites/svelte-5-preview/src/routes/docs/content/01-api/04-event-handlers.md

```diff
@@ -0,0 +1,180 @@
+---
+title: Event handlers
+---
+
+Event handlers have been given a facelift in Svelte 5. Whereas in Svelte 4 we use the `on:` directive to attach an event listener to an element, in Svelte 5 they are properties like any other:
+
+```diff
+<script>
+	let count = $state(0);
+</script>
+
+-<button on:click={() => count++}>
++<button onclick={() => count++}>
+	clicks: {count}
+</button>
+```
+
+Since they're just properties, you can use the normal shorthand syntax...
+
+```svelte
+<script>
+	let count = $state(0);
+
+	function onclick() {
+		count++;
+	}
+</script>
+
+<button {onclick}>
+	clicks: {count}
+</button>
+```
+
+...though when using a named event handler function it's usually better to use a more descriptive name.
+
+Traditional `on:` event handlers will continue to work, but are deprecated in Svelte 5.
+
+## Component events
+
+In Svelte 4, components could emit events by creating a dispatcher with [`createEventDispatcher`](https://svelte.dev/docs/svelte#createeventdispatcher).
+
+This function is deprecated in Svelte 5. Instead, components should accept _callback props_ ([demo](/#H4sIAAAAAAAAE41TS27bMBS8yoPaIHZry3IANYAiCe0Nuq-6UKTHmAhFCuKzHVfQvqsueoGuer8eISRF-YN40YUEcjgz5Bvy9QHjAnWQfOsDWTYYJMGXtg0WAR1aO9E7FIRmrtW2qyyS6qrjLeWFLIg3reoIvm6bFlinGrgNV3YSjrLbB0uyn0ACzX8gZPBeU0k4W8dzs5quTm4ytVJnK5kwnKyfzSHLoXcu5PQfM4hHV-IMZg7L4T6eT_aRWx0G-6_xms-Zbj3plpOtFa7cafp3hjjyMkN0hqluSwmVKLXOiuCxFEIpWQSg6SDQILoqBSbQR2G0hg9OPRRB_u_Pr5-mUqM1zn2CQuM1O6UaR_7990RecTa4bNwWLnQn8rW0SnPiSibA-AvWPhmBjBKfREGk2tNkz2vaJLCOohuPbJA_begCYkrS0h7eobv9ZIQvtCwFfzLbVSgJu2k_LnF59HmYjsYtZYk7Q9UJSCVxjNjWU1Do45tKoa6UUzEuR4jCO32pMAlN9FLyphzZDo7CWANT3b7s6gvV52c8sM48bQ3n-ugGevD3FYWfYBjhu_gCv4-nBRvP2coID-4F-5sxTdKomjOOdZBQt8Vhcewp-7L_t6lsr_Tge2AB_hHDYHun7VSrZ28753FrrlmCkpXg1XPWe_WQ-0G6Ghn5Fa7fYMj94Mh9W9D34RU91y9AMAQAAA==)):
+
+```svelte
+<script>
+	import Pump from './Pump.svelte';
+
+	let size = $state(15);
+</script>
+
+<Pump
+	inflate={() => {
+		size += 5;
+		if (size > 75) size = 0;
+	}}
+	deflate={() => {
+		if (size > 15) size -= 5;
+	}}
+/>
+
+{#if size >= 15}
```

Comment:
The example is playful but also hard to grok. It took a while for me to understand why the if is written they way it is (because you want boom to show between 0 and 15, which will happen when you inflate too much). I think it would be easier to have a separate button like "get new ballon" that resets the state to 0, and a separate flag like "ballonIntact" for the if/else block. Cramming that logic into one number is a bit confusing.